src/main/java/com/rtg/vcf/eval/VcfEvalCli.java

```diff
@@ -400,29 +403,84 @@ protected VcfEvalParams makeParams() throws IOException {
    * @throws IOException if a region based filter is specified but cannot be loaded
    */
   public static Set<RocFilter> getRocFilters(CFlags flags, List<RocFilter> fallbackRocFilters) throws IOException {
-    final Set<RocFilter> rocFilters = new LinkedHashSet<>(Collections.singletonList(RocFilter.ALL));  // We require the ALL entry for aggregate stats
-    if (flags.isSet(VcfEvalCli.ROC_SUBSET)) {
-      final List<?> values = flags.getValues(VcfEvalCli.ROC_SUBSET);
-      for (Object o : values) {
-        rocFilters.add(((RocFilterProxy) o).filter());
-      }
-    }
+    final LinkedHashMap<String, Set<RocFilter>> rocFiltersTypeMap = new LinkedHashMap<>();
+
+    final Set<RocFilter> rocExprFilters = new LinkedHashSet<>();
     if (flags.isSet(ROC_EXPR)) {
       final List<?> values = flags.getValues(VcfEvalCli.ROC_EXPR);
       for (Object o : values) {
         final String[] e = StringUtils.split((String) o, '=', 2);
         assert e.length == 2;
-        rocFilters.add(new ExpressionRocFilter(e[0], e[1]));
+        rocExprFilters.add(new ExpressionRocFilter(e[0], e[1]));
       }
     }
+    rocFiltersTypeMap.put("rocExprFilters", rocExprFilters);
+
+    final Set<RocFilter> rocRegionsFilters = new LinkedHashSet<>();
     if (flags.isSet(ROC_REGIONS)) {
       final List<?> values = flags.getValues(VcfEvalCli.ROC_REGIONS);
       for (Object o : values) {
         final String[] e = StringUtils.split((String) o, '=', 2);
         assert e.length == 2;
-        rocFilters.add(new RegionsRocFilter(e[0], new File(e[1])));
+        rocRegionsFilters.add(new RegionsRocFilter(e[0], new File(e[1])));
+      }
+    }
+    rocFiltersTypeMap.put("rocRegionsFilters", rocRegionsFilters);
+
+    // Split subset filters into genotype or variant flavors for intersecting in ROC_INTERSECT_MODE
```

Comment:
Yeah, this splitting of the `ROC_SUBSET` members into two subgroups is perhaps not obvious to the end user. Will think about this a little more.

Reply:
I think if you want to mix and match ROC stats across different types of subsets, it's natural to also split variant type and zygosity into separate dimensions for the subsetting. There will still be outputs for the larger categories, e.g. just Hets, just SNPs, etc. but also for the cross between them.